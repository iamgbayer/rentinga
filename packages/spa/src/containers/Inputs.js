import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import Store from 'store'
import media from 'styled-media-query'
import { withBreakpoints } from 'react-breakpoints'
import { useTranslation } from 'react-i18next'

import { Icon, Input, Select, Box } from 'components'

const Main = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;

  ${media.lessThan('small')`
    flex-direction: column;
  `}
`

const Secondary = styled(Box)`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`

export const Inputs = withBreakpoints(
  ({ compacted, breakpoints, currentBreakpoint }) => {
    const { filters } = Store.useStores()
    const { t } = useTranslation()
    const { bedroom, garage, price, type } = filters.state

    const isMobile = breakpoints[currentBreakpoint] === breakpoints.small

    const set = type => filter => filters.set(type)(filter)

    const props = {
      true: {
        secondary: {
          top: 30
        },
        select: {
          size: Select.size.default
        }
      },
      false: {
        secondary: {
          top: 50
        },
        select: {
          size: Select.size.box
        }
      }
    }

    return (
      <>
        <Main>
          <Input
            smooth={true}
            placeholder={t('home.search')}
            size={Input.size.large}
            right={isMobile ? 0 : 15}
            bottom={isMobile ? 15 : 0}
            full={isMobile}
            detailed={true}
          />

          <Select
            smooth={true}
            size={Select.size.large}
            full={isMobile}
            defaultValue={{ ...type, text: t(type.text) }}
            onChange={set('type')}
            options={[
              { value: 'apartamentos', text: t('home.type.apartments') },
              { value: 'casas', text: t('home.type.houses') },
              { value: 'kitnet', text: t('home.type.kitnet') }
            ]}
          />
        </Main>

        <Secondary data-testid="secondary" top={props[compacted].secondary.top}>
          <Select
            right={15}
            size={props[compacted].select.size}
            defaultValue={{ ...price, text: t(price.text) }}
            onChange={set('price')}
            icon={({ color, width, height }) => (
              <Icon name="dollar" color={color} width={width} height={height} />
            )}
            options={[
              { value: [0, 600], text: t('home.price.0,600') },
              { value: [601, 1000], text: t('home.price.601,1000') },
              { value: [1001, 1500], text: t('home.price.1001,1500') },
              { value: [1501, 3000], text: t('home.price.1501,3000') },
              { value: [3001, 5000], text: t('home.price.3001,5000') },
              { value: [5001, 50000], text: t('home.price.5001,50000') }
            ]}
          />

          <Select
            right={15}
            size={props[compacted].select.size}
            onChange={set('bedroom')}
            defaultValue={{ ...bedroom, text: t(bedroom.text) }}
            icon={({ color, width, height }) => (
              <Icon
                name="bedrooms"
                color={color}
                width={width}
                height={height}
              />
            )}
            options={[
              { value: [0, 0], text: t('home.bedroom.0,0') },
              { value: [1, 1], text: t('home.bedroom.1,1') },
              { value: [2, 2], text: t('home.bedroom.2,2') },
              { value: [3, 50], text: t('home.bedroom.3,50') },
              { value: [0, 50], text: t('home.bedroom.0,50') }
            ]}
          />

          <Select
            right={15}
            size={props[compacted].select.size}
            defaultValue={{ ...garage, text: t(garage.text) }}
            onChange={set('garage')}
            icon={({ color, width, height }) => (
              <Icon name="car" color={color} width={width} height={height} />
            )}
            options={[
              { value: [0, 0], text: t('home.garage.0,0') },
              { value: [1, 1], text: t('home.garage.1,1') },
              { value: [2, 50], text: t('home.garage.2,50') },
              { value: [0, 50], text: t('home.garage.0,50') }
            ]}
          />
        </Secondary>
      </>
    )
  }
)

Inputs.defaultProps = {
  compacted: false
}

Inputs.propTypes = {
  compacted: PropTypes.bool.isRequired
}
