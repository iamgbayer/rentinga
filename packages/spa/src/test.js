import React from 'react'
import { Router } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'
import { createMemoryHistory } from 'history'

import { Provider } from 'state'
import { Tokens } from 'components'

export const history = { ...createMemoryHistory('/'), push: jest.fn() }

export const withDependencies = Component => (
  <ThemeProvider theme={Tokens}>
    <Provider>
      <Router history={history}>{Component}</Router>
    </Provider>
  </ThemeProvider>
)
