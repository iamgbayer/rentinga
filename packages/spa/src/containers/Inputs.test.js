import React from 'react'
import { render } from '@testing-library/react'

import { Inputs } from './Inputs'
import { withDependencies } from 'test'

/**
 * @todo This import shouldn't be here, see rtl.config.js
 */
import 'jest-styled-components'

test('Should not expect visual regression', () => {
  const { container } = render(withDependencies(<Inputs />))

  expect(container.firstChild).toMatchSnapshot()
})

test('Should renders accordingly compacted property', () => {
  const { getByTestId, rerender } = render(withDependencies(<Inputs />))

  expect(getByTestId('secondary')).toHaveStyleRule('margin-top', '50px')

  rerender(withDependencies(<Inputs compacted={true} />))

  expect(getByTestId('secondary')).toHaveStyleRule('margin-top', '30px')
})
