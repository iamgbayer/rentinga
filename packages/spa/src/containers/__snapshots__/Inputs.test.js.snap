// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should not expect visual regression 1`] = `
.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  line-height: 0;
}

.c1 {
  margin-right: 15px;
  width: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 350px;
  height: 50px;
}

.c2 {
  border: none;
  width: 100%;
  font-size: 1.125rem;
  font-weight: 500;
  border-radius: 0.25rem;
  background-color: #fff;
  color: #2a3a3b;
  outline: none;
  padding: 0 15px;
  position: relative;
  z-index: 10;
  box-shadow: 0 2px 10px rgba(87,93,169,0.2);
}

.c3 {
  width: 60px;
  background-color: #3dd9d4;
  border-radius: 0.25rem;
  z-index: 5;
  position: absolute;
  top: 0;
  left: -3px;
  height: 50px;
}

.c4 {
  position: relative;
  width: 100%;
  max-width: 350px;
}

.c5 {
  width: 100%;
  padding: 0 16px;
  border-radius: 0.25rem;
  background-color: #fff;
  color: #2a3a3b;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 50px;
  box-shadow: 0 2px 10px rgba(87,93,169,0.2);
}

.c8 {
  width: 100%;
  top: calc(100% + 4px);
  position: absolute;
  border-radius: 0.25rem;
  overflow: hidden;
  list-style: none;
  z-index: 9000;
  box-shadow: 0 2px 10px rgba(87,93,169,0.2);
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 40px;
  padding-left: 14px;
  padding-right: 14px;
  cursor: pointer;
  color: #2a3a3b;
  background-color: #fff;
  font-weight: 400;
  font-size: 0.875rem;
}

.c9:hover {
  font-weight: 500;
}

.c6 {
  font-size: 90%;
  font-weight: 500;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0"
>
  <div
    class="c1"
    placeholder="Procurar"
  >
    <input
      class="c2"
    />
    <div
      class="c3"
    />
  </div>
  <div
    class="c4"
  >
    <div
      class="c5"
    >
      <span
        class="c6"
      >
        Apartamentos
      </span>
      <div
        class=""
        height="15"
        width="15"
      >
        <div
          class="c7"
        />
      </div>
    </div>
    <ul
      class="c8"
      style="opacity: 0; display: none;"
    >
      <li
        class="c9"
        value="apartamentos"
      >
        Apartamentos
      </li>
      <li
        class="c9"
        value="casas"
      >
        Casas
      </li>
      <li
        class="c9"
        value="kitnet"
      >
        Kitnet
      </li>
    </ul>
  </div>
</div>
`;
